processing/src/main/java/io/druid/query/extraction/MapLookupExtractor.java

```diff
@@ -72,13 +73,19 @@ public String apply(@NotNull String val)
   @Override
   public List<String> unapply(final String value)
   {
-    return Lists.newArrayList(Maps.filterKeys(map, new Predicate<String>()
-    {
-      @Override public boolean apply(@Nullable String key)
+    if (this.map instanceof HashBiMap) {
+      return Lists.newArrayList(((HashBiMap<String, String>) this.map).inverse().get(value));
```

Comment:
from the Interface docs:

```
   * @return the list of keys that maps to value or empty list.
   * Note that for the case of a none existing value in the lookup we have to cases either return an empty list OR list with null element.
   * returning an empty list implies that user want to ignore such a lookup value.
   * In the other hand returning a list with the null element implies user want to map the none existing value to the key null.
```

I'm actually not 100% sure which one should be done here, @b-slim do you have any insight for this?

Reply:
I've modified to match what the existing implementation and other interface implementations (I think) do.